src/runtimes/node/bundlers/esbuild/bundler.ts

```diff
@@ -29,45 +29,49 @@ const RESOLVE_EXTENSIONS = ['.js', '.jsx', '.mjs', '.cjs', '.ts', '.tsx', '.mts'
 /**
  * Our own `includedFiles` syntax is slightly different from what esbuild expects as `externals`.
  *
- * Turns !node_modules/test/** into ./node_modules/test/*
+ * Turns !node_modules/test/** into test
+ * and !lang/en.* into ./lang/en.*.
+ * ESBuild can't handle multiple globs in a single pattern, so we resolve them instead.
  */
-const includedFilesToEsbuildExternals = (
-  includedFiles: string[],
-  failOnDouble: boolean,
-  logger: Logger,
-  functionName: string,
-) =>
-  includedFiles
-    // only keep negated patterns
+const includedFilesToEsbuildExternals = async (includedFiles: string[], baseDir: string) => {
+  const exclusions = includedFiles
     .filter((pattern) => pattern.startsWith('!'))
-    // warn about invalid patterns
-    .filter((pattern) => {
-      const containsMultipleStars = pattern.indexOf('*') !== pattern.lastIndexOf('*')
+    .map((pattern) => pattern.slice(1))
+    // esbuild expects relative paths
+    .map((pattern) => `./${pattern}`)
 
-      if (containsMultipleStars) {
-        const warning = `Found invalid pattern in includedFiles: ${pattern}. Only one * is allowed when using esbuild. See https://ntl.fyi/esbuild-double-glob for more.`
+  // esbuild treats * the same as glob treats **, so this replacement is safe
+  const singleStar = exclusions.map((pattern) => pattern.replace('**', '*').replace(/\*(\\\*)+/, '*'))
```

Comment:
nit: I find this variable a bit misleading, because it doesn't necessarily have anything to do with single stars. It's just a version of `exclusions` with normalised paths. Could we do this in another `.map` in the `exclusions` declaration above, as opposed to creating a new variable?

Reply:
The idea is that this replaces all occurences of `**` with `*`, and all occurences of `*/*` with `*`. So `singleStar` is intended to mean "it only contains singular globs", not "it only contains one glob". I wanted the code to explain that somewhere, so I added the variable and the comment - but it reads like you interpreted it differently, so that didn't work. How would you phrase this?

Reply:
I would add a comment explaining what you said just now, but keep it as part of the group of maps above.

```js
  const exclusions = includedFiles
    .filter((pattern) => pattern.startsWith('!'))
    .map((pattern) => pattern.slice(1))
    // esbuild expects relative paths
    .map((pattern) => `./${pattern}`)
    // replace all occurences of ** with *, and all occurences of */* with *
    .map((pattern) => pattern.replace('**', '*').replace(/\*(\\\*)+/, '*'))
```